packages/server/lib/crons/deleteOldData.ts

```diff
@@ -0,0 +1,107 @@
+import * as cron from 'node-cron';
+import { errorManager, ErrorSourceEnum, deleteJobsByDate } from '@nangohq/shared';
+import { getLogger, metrics } from '@nangohq/utils';
+import tracer from 'dd-trace';
+import { setTimeout } from 'node:timers/promises';
+import type { Lock } from '@nangohq/kvstore';
+import { getLocking } from '@nangohq/kvstore';
+import db from '@nangohq/database';
+import { deleteExpiredPrivateKeys } from '@nangohq/keystore';
+import { envs } from '../env.js';
+import { deleteExpiredConnectSession } from '../services/connectSession.service.js';
+
+const logger = getLogger('Jobs.deleteOldData');
+
+const cronMinutes = envs.CRON_DELETE_OLD_DATA_EVERY_MIN;
+
+const limit = envs.CRON_DELETE_OLD_JOBS_LIMIT;
+const deleteJobsOlderThan = envs.CRON_DELETE_OLD_JOBS_MAX_DAYS;
+
+const deleteConnectionSessionOlderThan = envs.CRON_DELETE_OLD_CONNECT_SESSION_MAX_DAYS;
+const deletePrivateKeysOlderThan = envs.CRON_DELETE_OLD_PRIVATE_KEYS_MAX_DAYS;
+
+export function deleteOldData(): void {
```

Comment:
can we add a kill switch for every cron. ie: if  `cronMinutes == 0` nothing is started. Just in case